attalos/imgtxt_algorithms/infer.py

```diff
@@ -0,0 +1,117 @@
+                start_index = batch_num*batch_size
+                stop_index = min((batch_num+1)*batch_size, dataset.num_images)
+                image_features = dataset.image_feats[start_index:stop_index]
+                output_feats = model.predict_feats(sess, image_features)
+                output_file['preds'][start_index:stop_index] = output_feats
+
+        output_file.create_dataset('ids', data=dataset.image_ids)
+        output_file.close()
+
+
+def main():
+    import argparse
+
+    parser = argparse.ArgumentParser(description='Two layer linear regression')
+    parser.add_argument("image_feature_file_train",
+                        type=str,
+                        help="Image Feature file for the training set")
+    parser.add_argument("text_feature_file_train",
+                        type=str,
+                        help="Text Feature file for the training set")
+    parser.add_argument("output_hdf5_file",
+                        type=str,
+                        help="Output HDF5 file")
+    parser.add_argument("word_vector_file",
+                        type=str,
+                        help="Text file containing the word vectors")
+
+    # Optional Args
+    parser.add_argument("--learning_rate",
+                        type=float,
+                        default=0.001,
+                        help="Learning Rate")
+    parser.add_argument("--batch_size",
+                        type=int,
+                        default=128,
+                        help="Batch size to use for training")
+    parser.add_argument("--model_type",
+                        type=str,
+                        default="multihot",
+                        choices=['multihot', 'naivesum', 'wdv', 'negsampling', 'fast0tag'],
+                        help="Loss function to use for training")
+    parser.add_argument("--model_input_path",
+                        type=str,
+                        default=None,
+                        help="Model input path (to continue training)")
+
+    # new args
+    parser.add_argument("--hidden_units",
+                        type=str,
+                        default="200,200",
+                        help="Define a neural network as comma separated layer sizes")
+    parser.add_argument("--word_vector_type",
+                        type=str,
+                        choices=[t.name for t in WordVectorTypes],
+                        help="Format of word_vector_file")
+    parser.add_argument("--optim_words",
+                        action="store_true",
+                        default=False,
+                        help="If using negsampling model_type, use to jointly optimize words")
+    parser.add_argument("--joint_factor",
+                        type=float,
+                        default=1.0,
+                        help="Multiplier for learning rate in updating joint optimization")
+
+    args = parser.parse_args()
+    infer_dataset(args)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Created Issue #109 to track this. I think the changes are actually decently large so I might put it in a separate PR.